packages/cli/src/commands/configure/profile_controller.test.ts

```diff
@@ -218,6 +220,70 @@ void describe('profile controller', () => {
       );
     });
 
+    void it('throws error if config file already exists and is missing read permissions', async () => {
+      if (process.platform.startsWith('win')) {
+        // Windows does not have the same behavior when files are missing permissions
+        return;
+      }
+
+      const expectedErr = new AmplifyUserError('PermissionsError', {
+        message: `You do not have the permissions to read this file: ${configFilePath}.`,
+        resolution: `Ensure that you have the right permissions to read from ${configFilePath}.`,
+      });
+      chmod(configFilePath, 0o000, () => {
+        //do nothing
+        return;
+      });
```

Comment:
we should either use `chmodSync` from `fs` or `import { chmod } from fs/promises` and await it. 
Otherwise this code has a potential race condition (rare but still).